Changed GetEmployeeTeamSnapshot call to match new schema

nmcmahon-unosquare opened the pull request:
Created new query.
Added new models
Updated existing controller, mappers and models.

Jamie-Unosquare commented on v2/AdminCore.Services/DashboardService.cs:
```diff
@@ -163,14 +165,64 @@ private static EmployeeSnapshotDto CreateEmployeeSnapshotFromContract(Contract c
+      switch (@event.EventTypeId)
+      {
+        case 1:
+          return EmployeeLocationConstants.OnHoliday;
```
Small thing, change this to AnnualLeave for consistency with the rest of the project

nmcmahon-unosquare replied:
OnHoliday encompasses multiple holiday types (Annual Leave, Public Holiday, Mandatory leave.. etc..) and was explicitly requested by FE

Jamie-Unosquare commented on v2/AdminCore.Constants/EmployeeLocationConstants.cs:
```diff
@@ -0,0 +1,19 @@
+{
+  public class EmployeeLocationConstants
+  {
+    public static string OnHoliday = "On Holiday";
```
Small thing, change this to AnnualLeave for consistency with the rest of the project

Jamie-Higgins commented on v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs:
```diff
@@ -146,7 +146,7 @@ public void GetTeamEvents_WhenCalled_ReturnsTeamEvents()
       var dashboardEventModels = _fixture.CreateMany<DashboardEventViewModel>(numberOfEvents);
 
-      _dashboardService.GetTeamDashboardEvents(Arg.Is(employeeId), searchDate).Returns(eventsReturnedFromService);
+      //_dashboardService.GetTeamDashboardEvents(Arg.Is(employeeId), searchDate).Returns(eventsReturnedFromService);
```
Excuse me?!

Jamie-Unosquare commented on v2/AdminCore.WebApp/Mappings/WebMappingProfile.cs:
```diff
@@ -69,6 +71,14 @@ public WebMappingProfile()
       CreateMap<ContractDto, CreateContractViewModel>().ReverseMap();
       CreateMap<ContractDto, UpdateContractViewModel>().ReverseMap();
+
+      CreateMap<ClientSnapshotDto, ClientSnapshotViewModel>();
```
These 3 are unnecessary as ReverseMap includes the mapping for both directions

Jamie-Unosquare commented on v2/AdminCore.Services/Mappings/DashboardMapperProfile.cs:
```diff
@@ -0,0 +1,24 @@
+  {
+    public DashboardMapperProfile()
+    {
+      CreateMap<Client, ClientSnapshotDto>();
```
These 3 are unnecessary as ReverseMap includes the mapping for both directions

Jamie-Unosquare reviewed (changes requested):
Change constant name and remove unused mappings

nmcmahon-unosquare committed:
Changed GetEmployeeTeamSnapshot call to match new schema

Created new query.
Added new models
Updated existing controller, mappers and models.
